### PR TITLE
Remove `AccleratorConnector.num_ipus` and deprecate `Trainer.ipus`

### PR DESCRIPTION
**Pull request by DuYicong515**
## What does this PR do?
Part of #11449 : remove unused properties in accelerator_connector: num_ipus
Part of #12171 : deprecate device related properties in Trainer that is accessed from accelerator_connector: ipus

`AcceleratorConnector.num_ipus` was not meant to be public. It's removed from AcceleratorConnector and deprecated `Trainer.ipus` in favor of using `Trainer.num_devices if isinstance(Trainer.accelerator, IPUAccelerator) else 0`

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/accelerators/test_ipu.py [hunk 1]
```diff
@@ -561,7 +562,7 @@ def test_accelerator_ipu_with_devices():
 
     trainer = Trainer(accelerator="ipu", devices=8)
 
-    assert trainer.ipus == 8
+    assert trainer.num_devices == 8
     assert isinstance(trainer.strategy, IPUStrategy)
     assert isinstance(trainer.accelerator, IPUAccelerator)
```
Comment (krshrimali) [inline tests/accelerators/test_ipu.py:567]:
```suggestion
    assert isinstance(trainer.accelerator, IPUAccelerator)
    assert trainer.num_devices == 8
    assert isinstance(trainer.strategy, IPUStrategy)
```

Should the check for `trainer.accelerator` be before `trainer.num_devices == 8`?

File: tests/accelerators/test_ipu.py [hunk 2]
```diff
@@ -581,7 +582,7 @@ def test_accelerator_ipu_with_ipus_priority():
     with pytest.warns(UserWarning, match="The flag `devices=1` will be ignored,"):
         trainer = Trainer(accelerator="ipu", devices=1, ipus=ipus)
 
-    assert trainer.ipus == ipus
+    assert trainer.num_devices == ipus
```
Comment (krshrimali) [inline tests/accelerators/test_ipu.py:585]: This might sound repetitive, and I'm curious - whenever we are asserting `trainer.num_devices == <expected_ipus>` - should we also check if the accelerator was `IPUAccelerator` (in the tests)?

```suggestion
    assert isinstance(trainer.accelerator, IPUAccelerator)
    assert trainer.num_devices == ipus
```

**Review by krshrimali — APPROVED** [review]
LGTM, thanks @DuYicong515! Left minor comments, can you please take a look?

File: tests/accelerators/test_ipu.py (hunk 1 above)
Comment (rohitgr7) [reply tests/accelerators/test_ipu.py:567]: will it make any difference?

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: tests/accelerators/test_ipu.py (hunk 1 above)
Comment (DuYicong515) [reply tests/accelerators/test_ipu.py:567]: @krshrimali I think they'll work the same, we are asserting both the types and device numbers. 

If logically if asserting accelerator type first makes more sense to you, I can change that.

Comment (krshrimali) [reply tests/accelerators/test_ipu.py]: Sounds good to me, @DuYicong515 - thank you for the reasoning. Please don't change anything, it all looks good to me :)

File: tests/deprecated_api/test_remove_1-8.py [hunk 3]
```diff
@@ -962,3 +963,23 @@ def test_trainer_num_gpu_0(monkeypatch, gpus, expected_num_gpus, strategy):
         " Please use `Trainer.num_devices` instead."
     ):
         assert Trainer(gpus=gpus, strategy=strategy).num_gpus == expected_num_gpus
+
+
+@pytest.mark.parametrize(
+    ["trainer_kwargs", "expected_ipus"],
+    [
+        ({}, 0),
+        ({"devices": 1}, 0),
```
Comment (kaushikb11) [inline tests/deprecated_api/test_remove_1-8.py:972]: What's the purpose of these two sets of parameters? If we are testing for IPU Accelerator.

Comment (DuYicong515) [reply tests/deprecated_api/test_remove_1-8.py:972]: To test the branch that when IPU is not used Trainer.ipus should return 0

**Review by tchaton — APPROVED** [review]
LGTM !
